crates/router/src/types/api/webhooks.rs

```diff
@@ -101,23 +101,32 @@ pub trait IncomingWebhook: ConnectorCommon + Sync {
                 )
             })?
             .connector_webhook_details;
-        let merchant_secret = merchant_connector_webhook_details
-            .ok_or(errors::ConnectorError::WebhookSourceVerificationFailed)
-            .into_report()
-            .attach_printable_lazy(|| format!("Merchant Secret not configured {}", debug_suffix))?
-            .expose()
-            .parse_value::<MerchantConnectorWebhookDetails>("MerchantConnectorWebhookDetails")
-            .change_context_lazy(|| errors::ConnectorError::WebhookSourceVerificationFailed)
-            .attach_printable_lazy(|| {
-                format!(
-                    "Deserializing MerchantConnectorWebhookDetails failed {}",
-                    debug_suffix
-                )
-            })?
-            .merchant_secret
-            .expose();
+        let merchant_secret = match merchant_connector_webhook_details {
+            Some(merchant_connector_webhook_details) => Some(
+                merchant_connector_webhook_details
+                    .parse_value::<MerchantConnectorWebhookDetails>(
+                        "MerchantConnectorWebhookDetails",
+                    )
+                    .change_context_lazy(|| errors::ConnectorError::WebhookSourceVerificationFailed)
+                    .attach_printable_lazy(|| {
+                        format!(
+                            "Deserializing MerchantConnectorWebhookDetails failed {}",
+                            debug_suffix
+                        )
+                    })?
+                    .merchant_secret
+                    .expose(),
+            ),
+            None => None,
```

Comment:
return `Some("default_secret")` here itself?

Reply:
If I do that, I would have to do `merchant_secret.unwrap()` while returning.
This will cause `cargo clippy`  to fail